File: 2.add-two-numbers.py
#
# @lc app=leetcode id=2 lang=python3
#
# [2] Add Two Numbers
#

# @lc code=start
# Definition for singly-linked list.
<<<<<<< HEAD
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        result = ListNode(0)
        carry = 0
        r=result
        while l1 and l2:
            current = ListNode(0)
            current.val = (l1.val+l2.val+carry) // 10
            carry  = (l1.val+l2.val + carry )%10
            l1 = l1.next
            l2 = l2.next
            current.next = ListNode(0)
            current = current.next
            r = r.next
            r=current
            
        if l1:
            current = l1
        if l2:
            current = l2
        return result

        
=======
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        if l1.val == 0 and not l1.next:
            return l2
        if l2.val == 0 and not l2.next:
            return l1
        carry = 0
        l3 = ListNode((l1.val+l2.val)%10)
        carry = (l1.val + l2.val)//10
        current = l3
        l1 = l1.next
        l2 = l2.next
        while l1 and l2:
            node = ListNode((l1.val + l2.val + carry)%10)
            carry = (l1.val + l2.val + carry) // 10
            l1=l1.next
            l2 = l2.next   
            current.next = node 
            current = current.next
        remain = l1 if l1 else (l2 if l2 else None)

        while remain and carry:
            node = ListNode((remain.val + carry)%10)
            carry = (remain.val + carry) // 10
            remain=remain.next
            current.next = node 
            current = current.next
        if carry:
            current.next = ListNode(carry)
        if remain:
            current.next = remain   

        return l3

            

        
        
>>>>>>> RFC:change mac
# @lc code=end

File: 2.add-two-numbers.go
/*
 * @lc app=leetcode id=2 lang=golang
 *
 * [2] Add Two Numbers
 */
package main

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	sum := &ListNode{}
	head := sum
	carry := 0
	for l1 != nil && l2 != nil {

		val := (l1.Val + l2.Val + carry) % 10
		carry = (l1.Val + l2.Val + carry) / 10
		l1 = l1.Next
		l2 = l2.Next

		node := ListNode{Val: val}
		head.Next = &node

		head = head.Next
	}
	if l1 != nil {
		l2 = l1
	}

	for l2 != nil {
		val := (l2.Val + carry) % 10
		carry = (l2.Val + carry) / 10
		head.Next = &ListNode{Val: val}
		head = head.Next
		l2 = l2.Next
	}
	if carry != 0 {
		head.Next = &ListNode{Val: carry}
	}
	return sum.Next
}

// @lc code=end
